src/services/workflow/WorkflowService.ts

```diff
@@ -0,0 +1,85 @@
+import { IStatus, IWorkflow, Transition } from "common/models";
+import { injectable } from "inversify";
+import { IWorkflowService } from "services/workflow/model";
+import "reflect-metadata";
+
+@injectable()
+export class WorkflowService implements IWorkflowService {
+  parseTransitionsToKeyValuePair(transitions: Transition): Array<string> {
+    const pairs: Array<string> = [];
+    transitions.forEach((value, key) => {
+      const validStatuses = value.reduce(
+        (acc, status, index) =>
+          (acc += status.name + (index !== value.length - 1 ? ", " : "")),
+        ""
+      );
+      pairs.push(key.name + ";" + validStatuses);
+    });
+
+    return pairs;
+  }
+
+  getAllValuesOfTransitionsMap(transitions: Transition): Array<Array<IStatus>> {
+    const statusesInMap = transitions.values();
+    const statusFieldTransitions: Array<Array<IStatus>> = [];
+
+    for (let i = 0; i < transitions.size; i++) {
+      statusFieldTransitions.push(statusesInMap.next().value);
+    }
+
+    return statusFieldTransitions;
+  }
+
+  parseTransitionsToStringArray(
+    workflow?: IWorkflow
+  ): Array<Array<string>> | undefined {
+    if (!workflow) {
```

Comment:
If pass nothing it will do nothing. It is hard to name a method that does that. Workflow can be required and the check, if the workflow exists, can be done where the method is used

Reply:
done